### ISSUE TITLE
Errors when handling nested directories with spaces

### ISSUE THREAD

**Issue by andkrau** [mar10/wsgidav#228]
**I get the following error when using the OPTIONS method prior to uploading a file to a directory with spaces that also has a parent directory with spaces:**
```
13:28:39.218 No authorization required for OPTIONS method
13:28:39.220 Raising DAVError 404 Not Found: /Black/Directory%20With%20Spaces/Subdirectory%20With%20Spaces
13:28:39.220 re-raising DAVError(404 Not Found: /Black/Directory%20With%20Spaces/Subdirectory%20With%20Spaces)
13:28:39.222 caught DAVError(404 Not Found: /Black/Directory%20With%20Spaces/Subdirectory%20With%20Spaces)
13:28:39.223 192.168.1.201 - (anonymous) - [2021-08-22 18:28:39] "OPTIONS /Black/Directory%20With%20Spaces/Subdirectory%20With%20Spaces" connection="TE, Keep-Alive", agent="WinSCP/5.19.2 neon/0.31.2", elap=0.057sec -> 404 Not Found
13:28:39.224 <2220> ---OPTIONS  Response(404 Not Found): ---
```
**No error when attempting to upload the file to the parent directory (subsequent attempts with the child directory will now succeed):**
```
13:35:39.443 No authorization required for OPTIONS method
13:35:39.446 192.168.1.201 - (anonymous) - [2021-08-22 18:35:39] "OPTIONS /Black/Directory%20With%20Spaces" connection="TE, Keep-Alive", agent="WinSCP/5.19.2 neon/0.31.2", elap=0.049sec -> 200 OK
13:35:39.447 <1316> ---OPTIONS  Response(200 OK): ---
```
EDIT: Whoops, looks like the "unquote_path_info" option resolves this. WinSCP and CarotDAV now work perfectly when running WsgiDAV on Windows. #8

**Comment by mar10** [comment]
Which server did you use (cheroot,ext-wsgiutils,gevent,gunicorn,paste,uvicorn,wsgiref)?

**Comment by andkrau** [comment]
This was with cheroot.

**Comment by mar10** [comment]
The default for `hotfixes.unquote_path_info` is false, since it fixes #8 

I could not reproduce this issue with WsgiDAV 4.0.pre on Windows 10 / File Explorer.

So it *may* be fixed with 4.0 (or may be specific to some clients like WinSCP).
Closing for now, but it would be great if you could feedback, once you upgrade to 4.0.

**Comment by mar10** [comment]
just tried with WinSCP and it also worked for me:
![image](https://user-images.githubusercontent.com/41663/143690034-b416841a-1deb-4670-ba38-d6a88f6457ce.png)
